### PR TITLE
Mock entity service in component unit tests

### PR DESCRIPTION
**Pull request by hdurix**
-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/jhipster-vuejs/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/jhipster-vuejs/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by hdurix** [comment]
This PR is to mock the service layer in component unit tests instead of mocking axios (the axios is already tested by service unit test).

The only problem is that the unit tests are now longer to execute! It seems that babel is optimizing `sinon` dependency. If anyone has an idea why babel is trying to optimize what it is in `node_modules`, please give me a hint.

This PR is in draft status to check if we can configure babel correctly before merging it.

File: generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity-management.component.spec.ts.ejs [hunk 1]
```diff
@@ -86,49 +81,49 @@ describe('Component Tests', () => {
...
+      expect(<%= entityInstance %>ServiceStub.retrieve.called).toBeFalsy();
     });
     <%_ } _%>
 
     it('should re-initialize the page', async () => {
       // GIVEN
-      mockedAxios.get.mockReset();
-      mockedAxios.get.mockReturnValue(Promise.resolve({headers: {}, data: [{id: <%- tsKeyId %>}]}));
+      <%= entityInstance %>ServiceStub.retrieve.reset();
+      <%= entityInstance %>ServiceStub.retrieve.resolves({ headers: {}, data: [{ id: 123 }] });
```
Comment (nonomoho) [inline generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity-management.component.spec.ts.ejs]: Why are you using `123` and not `tsKeyId` here?

Comment (hdurix) [reply generators/entity-client/templates/vue/src/test/javascript/spec/app/entities/entity-management.component.spec.ts.ejs]: Good catch!

**Comment by Gnuk** [comment]
I approve the review but I think we'll need to know the problem with sinon optimization before merge.

**Comment by hdurix** [comment]
Before this MR: 15s
After this MR: 40s

But if we find the issue, maybe it will be quicker than before :)

**Comment by hdurix** [comment]
I found it! Now tests are taking 10s.
I'll add the fix

**Comment by hdurix** [comment]
Travis and Azure tests are failing because of jhipster info step, `semver` seems now missing (I think this dependency has been remove a long time ago in this blueprint). Any idea why this is failing now @pascalgrimaud @sahbi-ktifa? Is it linked to this PR?

**Comment by pascalgrimaud** [comment]
@Hippolyte : it failed because of new release of generator-jhipster. We need to upgrade to 6.1.0

**Comment by hdurix** [comment]
Ok thanks @pascalgrimaud, let me do a new PR then.

**Comment by hdurix** [comment]
Just rebased, can now be merged.

**Comment by hdurix** [comment]
As discussed with @sahbi-ktifa I'm merging it.

**Pull request merged**